content/docs/app-developer-guide/mouting-volumes.md

```diff
@@ -0,0 +1,81 @@
++++
+title="Mounting Volumes"
+weight=1
+summary="Volumes are a mechanism for both supplying and persisting data generated by build containers."
++++
+
+{{< param "summary" >}}
+
+### Mounting Volumes (`--volume`)
+
+The `--volume` parameter must be one of the following:
+
+ - A volume mount.
+ - A comma separated list of volume mounts.
+
+Each volume mount has the following format:
+```
+<host path>:<target path>[:<mode>]
+```
+
+##### `<host path>`
+This is the name of the volume, it is unique on a given host machine.
+
+##### `<target path>`
+The path where the file or directory is available in the container.
+
+##### `[:<mode>]` 
+An optional comma separated list of mount options. If no options are provided, the read-only option will be used.
+A mount option must be one of the following:
+  - `ro`, volume contents are read-only.
+  - `rw`, volume contents are readable and writeable.
+  - `volume-opt=key=value`, can be specified more than once, takes a key-value pair consisting of the option name and its value.
+
+
+#### Examples:
+For the following examples we will use our [samples][samples] repo for simplicity.
+
+Here we bind mount a local folder into our container and display its contents during 
+a `pack build`.
+
+##### Linux container Example:
+
+```bash
+# create a new docker volume
+docker volume create test-volume
+
+# put a text file in it
+docker run --rm --name=populate-volume \
+    --volume test-volume:/tmp/volume:rw \
+    bash \
+    bash -c 'echo "Hello from a volume!" > /tmp/volume/volume_contents.txt'
+
+# pack build using volume! 
+pack build volume-example \
+    --builder cnbs/sample-builder:bionic \
+    --buildpack samples/buildpacks/hello-world \
+    --path samples/apps/bash-script \
+    --volume test-volume:/platform/volume:ro
+```
+
+##### Linux container on Windows host Example:
+
+```
+# create a new docker volume
+docker volume create test-volume
+
+# put a text file in it
+docker run --rm --name=populate-volume `
+    --volume test-volume:/tmp/volume:rw `
+    bash `
+    bash -c "echo 'Hello from a volume!' > /tmp/volume/volume_contents.txt"
+
+#
+pack build volume-example `
+    --builder cnbs/sample-builder:bionic `
+    --buildpack .\samples\buildpacks\hello-world.tgz `
```

Comment:
If we use the id of the buildpack instead of a path to a prepackaged version of the buildpack I wonder if we need to differentiate between Windows and Linux hosts.

```suggestion
    --buildpack samples/hello-world@0.0.1 `
```